common/gossmap.c

```diff
@@ -835,3 +848,70 @@ u8 *gossmap_node_get_announce(const tal_t *ctx,
 	map_copy(map, n->nann_off, msg, len);
 	return msg;
 }
+
+/* BOLT #7:
+ * 1. type: 256 (`channel_announcement`)
+ * 2. data:
+ *     * [`signature`:`node_signature_1`]
+ *     * [`signature`:`node_signature_2`]
+ *     * [`signature`:`bitcoin_signature_1`]
+ *     * [`signature`:`bitcoin_signature_2`]
+ *     * [`u16`:`len`]
+ *     * [`len*byte`:`features`]
+ *     * [`chain_hash`:`chain_hash`]
+ *     * [`short_channel_id`:`short_channel_id`]
+ *     * [`point`:`node_id_1`]
+ *     * [`point`:`node_id_2`]
+ */
+int gossmap_chan_has_feature(const struct gossmap *map,
+			     const struct gossmap_chan *c,
+			     int fbit)
+{
+	/* Note that first two bytes are message type */
+	const size_t feature_len_off = 2 + (64 + 64 + 64 + 64);
+	size_t feature_len;
+
+	feature_len = map_be16(map, c->cann_off + feature_len_off);
+
+	if (map_feature_set(map, OPTIONAL_FEATURE(fbit),
+			    c->cann_off + feature_len_off + 2, feature_len))
+		return OPTIONAL_FEATURE(fbit);
+	if (map_feature_set(map, COMPULSORY_FEATURE(fbit),
```

Comment:
We could let `map_feature_set` return both optional and compulsory flags, in order to avoid calling it twice.

Reply:
Yes, it's fairly cheap, but since this is the only way we call it...